### ISSUE TITLE
Obtaining cached files via `SearchResult.download_all()` is unexpectedly slow

### ISSUE THREAD

**Issue by barentsen** [KeplerGO/lightkurve#914]
I find that loading 10 cached light curves using `download_all` is very slow.

The first call to `download_all` is of course very slow (25.7s) because 10 light curves have to be downloaded.  For example:

```python
>>> import lightkurve as lk
>>> search = lk.search_lightcurve("Pi Men", sector=1, radius=6000, limit=10)
>>> %time collection = search.download_all()
CPU times: user 4.41 s, sys: 344 ms, total: 4.75 s
Wall time: 25.7 s
```

This is entirely expected.  However, subsequent calls should be blazing fast because the files are now cached on disk.  Instead, they still take 7-8 seconds of wall time to run, which is unexpected:

```python
>>> %time collection = search.download_all()
CPU times: user 3.49 s, sys: 103 ms, total: 3.59 s
Wall time: 6.86 s
>>> %time collection = search.download_all()
CPU times: user 4.55 s, sys: 169 ms, total: 4.72 s
Wall time: 8.71 s
```

Some detailed profiling is required to understand why obtaining *cached* files using `download_all()` is so slow!

**Comment by barentsen** [comment]
I line-profiled `download_all()` and concluded that the majority of the time appears to be lost inside the function that reads and parses a generic light curve FITS file, i.e. `lightkurve.io.generic.read_generic_lightcurve()`.  In other words, the slow performance does not appear to be due to the download caching mechanism.

We can likely optimize this function:

```python
>>> %lprun -f lk.io.generic.read_generic_lightcurve search[0].download()
Timer unit: 1e-06 s

Total time: 0.701839 s
File: /Users/gb/dev/lightkurve/lightkurve/io/generic.py
Function: read_generic_lightcurve at line 16

Line #      Hits         Time  Per Hit   % Time  Line Contents
==============================================================
    16                                           def read_generic_lightcurve(filename, flux_column,
    17                                                                       quality_column='quality',
    18                                                                       cadenceno_column='cadenceno',
    19                                                                       centroid_col_column='mom_centr1',
    20                                                                       centroid_row_column='mom_centr2',
    21                                                                       time_format=None,
    22                                                                       ext=1):
    23                                               """Generic helper function to convert a Kepler ot TESS light curve file
    24                                               into a generic `LightCurve` object.
    25                                               """
    26         1          5.0      5.0      0.0      if isinstance(filename, fits.HDUList):
    27                                                   hdulist = filename  # Allow HDUList to be passed
    28                                               else:
    29         1       3973.0   3973.0      0.6          hdulist = fits.open(filename)
    30                                           
    31                                               # Raise an exception if the requested extension is invalid
    32         1          4.0      4.0      0.0      if isinstance(ext, str):
    33                                                   validate_method(ext, supported_methods=[hdu.name.lower() for hdu in hdulist])
    34         1       2298.0   2298.0      0.3      hdu = hdulist[ext]
    35         1     217947.0 217947.0     31.1      tab = Table.read(hdu, format='fits')
    36                                           
    37                                               # Make sure the meta data also includes header fields from extension #0
    38         1       6161.0   6161.0      0.9      tab.meta.update(hdulist[0].header)
    39                                           
    40         1         56.0     56.0      0.0      tab.meta = {k : v for k, v in tab.meta.items()}
    41                                           
    42                                               # Some KEPLER files used to have a T column instead of TIME.
    43         1         12.0     12.0      0.0      if "T" in tab.colnames:
    44                                                   tab.rename_column("T", "TIME")
    45                                           
    46        21         51.0      2.4      0.0      for colname in tab.colnames:
    47                                                   # Ensure units have the correct astropy format
    48        20     309593.0  15479.6     44.1          if tab[colname].unit == 'e-/s':
    49         6       3165.0    527.5      0.5              tab[colname].unit = 'electron/s'
    50        20     119338.0   5966.9     17.0          if tab[colname].unit == 'pixels':
    51         2        147.0     73.5      0.0              tab[colname].unit = 'pixel'
    52                                           
    53                                                   # Rename columns to lowercase
    54        20       2031.0    101.5      0.3          tab.rename_column(colname, colname.lower())
    55                                           
    56                                               # We *have* to remove rows with TIME=NaN because the Astropy Time
    57                                               # object does not support the presence of NaNs.
    58                                               # Fortunately, such rows are always bad data.
    59         1       1004.0   1004.0      0.1      nans = np.isnan(tab['time'].data)
    60         1         37.0     37.0      0.0      if np.any(nans):
    61         1         80.0     80.0      0.0          log.debug('Ignoring {} rows with NaN times'.format(np.sum(nans)))
    62         1       3080.0   3080.0      0.4      tab = tab[~nans]
    63                                           
    64                                               # Prepare a special time column
    65         1          3.0      3.0      0.0      if not time_format:
    66                                                   if hdu.header.get('BJDREFI') == 2454833:
    67                                                       time_format = 'bkjd'
    68                                                   elif hdu.header.get('BJDREFI') == 2457000:
    69                                                       time_format = 'btjd'
    70                                                   else:
    71                                                       raise ValueError(f"Input file has unclear time format: {filename}")
    72         2       2507.0   1253.5      0.4      time = Time(tab['time'].data,
    73         1        116.0    116.0      0.0                  scale=hdu.header.get('TIMESYS', 'tdb').lower(),
    74         1          2.0      2.0      0.0                  format=time_format)
    75         1         28.0     28.0      0.0      tab.remove_column('time')
    76                                           
    77                                               # For backwards compatibility with Lightkurve v1.x,
    78                                               # we make sure standard columns and attributes exist.
    79         1          2.0      2.0      0.0      if 'flux' not in tab.columns:
    80         1        365.0    365.0      0.1          tab.add_column(tab[flux_column], name="flux", index=0)
    81         1          4.0      4.0      0.0      if 'flux_err' not in tab.columns and f"{flux_column}_err" in tab.columns:
    82         1        368.0    368.0      0.1          tab.add_column(tab[f"{flux_column}_err"], name="flux_err", index=1)
    83         1          3.0      3.0      0.0      if 'quality' not in tab.columns and quality_column in tab.columns:
    84                                                   tab.add_column(tab[quality_column], name="quality", index=2)
    85         1          2.0      2.0      0.0      if 'cadenceno' not in tab.columns and cadenceno_column in tab.columns:
    86                                                   tab.add_column(tab[cadenceno_column], name="cadenceno", index=3)
    87         1          3.0      3.0      0.0      if 'centroid_col' not in tab.columns and centroid_col_column in tab.columns:
    88         1        255.0    255.0      0.0          tab.add_column(tab[centroid_col_column], name="centroid_col", index=4)
    89         1          2.0      2.0      0.0      if 'centroid_row' not in tab.columns and centroid_row_column in tab.columns:
    90         1        262.0    262.0      0.0          tab.add_column(tab[centroid_row_column], name="centroid_row", index=5)
    91                                           
    92         1        161.0    161.0      0.0      tab.meta['LABEL'] = hdulist[0].header.get('OBJECT')
    93         1        120.0    120.0      0.0      tab.meta['MISSION'] = hdulist[0].header.get('MISSION', hdulist[0].header.get('TELESCOP'))
    94         1         68.0     68.0      0.0      tab.meta['RA'] = hdulist[0].header.get('RA_OBJ')
    95         1         67.0     67.0      0.0      tab.meta['DEC'] = hdulist[0].header.get('DEC_OBJ')
    96         1          4.0      4.0      0.0      tab.meta['FILENAME'] = filename
    97                                           
    98         1      28515.0  28515.0      4.1      return LightCurve(time=time, data=tab)
```

**Comment by barentsen** [comment]
This has been resolved by #915, which is now merged! 🥳
